cmake/strip_binary.cmake

```diff
@@ -21,7 +21,7 @@ macro(clickhouse_strip_binary)
        COMMAND cp "${STRIP_BINARY_PATH}" "${STRIP_DESTINATION_DIR}/bin/${STRIP_TARGET}"
        COMMAND "${OBJCOPY_PATH}" --only-keep-debug --compress-debug-sections "${STRIP_DESTINATION_DIR}/bin/${STRIP_TARGET}" "${STRIP_DESTINATION_DIR}/lib/debug/bin/${STRIP_TARGET}.debug"
        COMMAND chmod 0644 "${STRIP_DESTINATION_DIR}/lib/debug/bin/${STRIP_TARGET}.debug"
-       COMMAND "${STRIP_PATH}" --remove-section=.comment --remove-section=.note "${STRIP_DESTINATION_DIR}/bin/${STRIP_TARGET}"
+       COMMAND "${STRIP_PATH}" "${STRIP_DESTINATION_DIR}/bin/${STRIP_TARGET}"
```

Comment:
I'd leave `--remove-section=.comment` to have it close to a former debian stripper

Reply:
What do you mean by "former debian stripper"?

(Another approach would be to embed the hash as ".ClickHouse" (instead of ".note.ClickHouse.hash") and to continue remove ".note" and ".comment". But I did not really understand what ".note" and ".comment" were used for except the hash.)

Reply:
We more or less reimplemented the debian debug symbols stripper in cmake, see https://github.com/ClickHouse/ClickHouse/pull/35196

Maybe, using `.ClickHouse.hash` or `.diagnostics.ClickHouse.hash` is a good idea, but it's worth reading if it's fine.

Reply:
Stripping in Debian: www.debian.org/doc/debian-policy/ch-files.html

So indeed, they explicitly remove `.note` and `.comment`. The real question is what `.note` and `.comment` actually store. These sections are non-standard and I found only vague information, e.g.
- https://lists.debian.org/debian-policy/1998/03/msg00256.html - a 1998 (!) mailthread
- https://jerry153fish.github.io/2017/10/11/ELF-format.html
- https://stackoverflow.com/questions/9401217/is-the-elf-notes-section-really-needed

In sum, they store the compiler version information or placeholders which seem to be non-mandatory for debugging. Also in some cases, both sections can be quite large, however, in ClickHouse, I found them actually very small. But maybe this also depends on the compiler.

Anyways, perhaps it's a good thing to stay compatible with Debian as long as we document it. Will push a follow-up in this PR.

Reply:
And another LOL:
```
./clickhouse hash-binary  # returns C66066CFE511A7C34A9A0C6281F44C21

# add the hash to the binary (happens for official builds)
objcopy --add-section .ClickHouse.hash=hash clickhouse

./clickhouse hash-binary  # returns 37337872D77DD22744223DDB5E464913

./clickhouse server
[...]
<Error> Application: DB::Exception: Calculated checksum of the ClickHouse binary (37337872D77DD22744223DDB5E464913) does not correspond to the reference checksum stored in the binary (C66066CFE511A7C34A9A0C6281F44C21). It may indicate one of the following: - the file /home/ubuntu/repo/ClickHouse1/build/programs/clickhouse-bak was changed just after startup; - the file /home/ubuntu/repo/ClickHouse1/build/programs/clickhouse-bak is damaged on disk due to faulty hardware; - the loaded executable is damaged in memory due to faulty hardware; - the file /home/ubuntu/repo/ClickHouse1/build/programs/clickhouse-bak was intentionally modified; - logical error in code.
[...]
```
I really wonder if that hashsum feature was ever verified to work ;-)

Reply:
Yes, it is verified to work.